### ISSUE TITLE
Passing credentials to instagram node

### ISSUE THREAD

**Issue by ramajd** [node-red/node-red-web-nodes#131]
I am using node-red as backend of my application. I pass flows to node-red using runtime api and with POST request to **/flows**. 

I authenticate my app to instagram and pass received token to node-red like following array: 
```json
[
    {
        "type": "instagram",
        "instagram": "",
        "inputType": "like",
        "outputType": "link",
        "name": "",
        "x": 123,
        "y": 321,
        "wires": [
            []
        ],
        "credentials": {
            "client_id": "my_client_id_from_instagram_panel",
            "client_secret": "my_client_secret_from_instagram_panel",
            "redirect_uri": "http://127.0.0.1:7070/red/instagram-credentials/auth/callback",
            "code": "auto_generated_hash_code",
            "username": "my_instagram_usename",
            "access_token": "received_token_from_instagram"
        }
    }
]
```
but when I POST this array to **/flows** url of api. the imported node doesnt have any credentials and cant access instagram images.
is there any solution for this matter ?

**Comment by knolleary** [comment]
That is not a valid configuration for the instagram node - there are a number of problems.

It is missing an `id` property - this should be a unique identifier for the node.

The `instagram` property is blank. This should be set to the id of an `instagram-credentials` node. It is this `instagram-credentials` node that defines the credentials properties used by the instagram nodes.

That said, I'm not sure it is possible to import a full-configured instagram-credentials node. If you look at what credentials it declares (https://github.com/node-red/node-red-web-nodes/blob/master/instagram/instagram.js#L367), you'll see it isn't the full set that the node actually uses. These are the ones that the user sets via the editor - it is assumed the node will always be configured by the editor which does the oauth flow under the covers to do the necessary token exchange and stores the results directly into credentials.

In summary, whilst your example flow isn't valid, the node needs to be updated to support this sort of import.

**Comment by ramajd** [comment]
my mistake, missing parts of node, like id, was added automatically by application. 
so if I had understood correctly, current version of Instagram node doesn't support passing credentials through run time API, is it right ?

**Comment by knolleary** [comment]
correct - the node needs to be configured via the editor UI to properly create the credentials.

**Comment by ramajd** [comment]
OK thanks for your assist.
